gpflow/conditionals/util.py

```diff
@@ -206,14 +249,18 @@ def independent_interdomain_conditional(Kmn,
     # compute the covariance due to the conditioning
     if full_cov and full_output_cov:
         fvar = Knn - tf.tensordot(Ar, Ar, [[0, 1], [0, 1]])  # [N, P, N, P]
+        cov_shape = "NPNP"
     elif full_cov and not full_output_cov:
         At = tf.reshape(tf.transpose(Ar), (P, N, M * L))  # [P, N, L]
         fvar = Knn - tf.linalg.matmul(At, At, transpose_b=True)  # [P, N, N]
+        cov_shape = "PNN"
     elif not full_cov and full_output_cov:
         At = tf.reshape(tf.transpose(Ar, [2, 3, 1, 0]), (N, P, M * L))  # [N, P, L]
         fvar = Knn - tf.linalg.matmul(At, At, transpose_b=True)  # [N, P, P]
+        cov_shape = "NPP"
     elif not full_cov and not full_output_cov:
         fvar = Knn - tf.reshape(tf.reduce_sum(tf.square(A), [0, 1]), (N, P))  # Knn: [N, P]
+        cov_shape = "NP"
```

Comment:
rename `cov_shape` to `intended_cov_shape` or similar, to make it clear it's the intended not necessarily actual shape